code/integration-tests/src/androidTest/java/com/adobe/marketing/mobile/integration/core/RulesEngineIntegrationTests.kt

```diff
@@ -0,0 +1,277 @@
+        const val TEST_APP_ID = "appId"
+        const val TEST_CONFIG_URL = "https://assets.adobedtm.com/${TEST_APP_ID}.json"
+        const val TEST_RULES_RESOURCE = "rules_dispatch_consequence.zip"
+        const val WAIT_TIME_MILLIS_LONG = 500L
+        const val WAIT_TIME_MILLIS_SHORT = 100L
+    }
+
+    @Before
+    fun setup() {
+        val appContext =
+            InstrumentationRegistry.getInstrumentation().targetContext.applicationContext as Application
+
+        // Setup with only configuration extension
+        SDKHelper.resetSDK()
+
+        val initializationLatch = CountDownLatch(1)
+        val configUrlValidationLatch = CountDownLatch(1)
+        val rulesUrlValidationLatch = CountDownLatch(1)
+
+        // Sets up a network service to respond with a mock config and rules
+        setupNetworkService(
+            mapOf(
+                "global.privacy" to "optedin",
+                "rules.url" to "https://rules.com/rules.zip"
+            ),
+            TEST_RULES_RESOURCE
+        ) {
+            when (it) {
+                TEST_CONFIG_URL -> configUrlValidationLatch.countDown()
+                TEST_RULES_URL -> rulesUrlValidationLatch.countDown()
+            }
+        }
+
+        MobileCore.setApplication(appContext)
+        MobileCore.setLogLevel(LoggingMode.VERBOSE)
+        MobileCore.start {
+            initializationLatch.countDown()
+        }
+        MobileCore.configureWithAppID(TEST_APP_ID)
+
+        assertTrue(initializationLatch.await(WAIT_TIME_MILLIS_LONG, TimeUnit.MILLISECONDS))
+        // Validate that the configuration url is hit
+        assertTrue(configUrlValidationLatch.await(WAIT_TIME_MILLIS_LONG, TimeUnit.MILLISECONDS))
+        // Validate that the rules url from config is hit
+        assertTrue(rulesUrlValidationLatch.await(WAIT_TIME_MILLIS_LONG, TimeUnit.MILLISECONDS))
+    }
+
+    @Test
+    fun testEventDispatchConsequence() {
+        val capturedEvents = mutableListOf<Event>()
+
+        val eventLatch = CountDownLatch(1)
+        MobileCore.registerEventListener(
+            "test.type.consequence",
+            "test.source.consequence",
+            object : AdobeCallbackWithError<Event> {
+                override fun call(value: Event) {
+                    capturedEvents.add(value)
+                    eventLatch.countDown()
+                }
+
+                override fun fail(error: AdobeError?) {
+                    eventLatch.countDown()
+                }
+            })
+
+        val event = Event.Builder("Test Event Trigger", "test.type.trigger", "test.source.trigger")
+            .setEventData(
+                mapOf("xdm" to "test data")
+            )
+            .build()
+        MobileCore.dispatchEvent(event)
+
+        assertTrue(eventLatch.await(WAIT_TIME_MILLIS_SHORT, TimeUnit.MILLISECONDS))
+        assertTrue(capturedEvents.size == 1)
+        assertEquals("test data", capturedEvents[0].eventData?.get("xdm"))
```

Comment:
nit: Let us assert the entire map as it makes it more readable. 
        assertEquals(mapOf("xdm" to "test data"), capturedEvents[0].eventData)